### PR TITLE
SonarQube on App Service on Linux with Azure SQL Database

### PR DESCRIPTION
**Pull request by DanielLarsenNZ**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by DanielLarsenNZ** [comment]
Travis error: `Requested features are not supported in region`
This works in `westus`. Which region is Travis using?

**Comment by DanielLarsenNZ** [comment]
@almrangers - would you mind reviewing `sonarqube-webapp-linux-azuresql/azuredeploy.json` ?

**Comment by DanielLarsenNZ** [comment]
Hey @krnese do you know a maintainer who can switch the Travis build to use westus? The build is failing because App Services on Linux is not supported in the region that it is using.

**Comment by DanielLarsenNZ** [comment]
Waiting on #3743 to be resolved.

File: sonarqube-webapp-linux-azuresql/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        "P2",
+        "P3"
+      ],
+      "metadata": {
+        "description": "Describes the performance level for Edition"
+      }
+    }
+  },
+  "variables": {
+    "hostingPlanName": "[parameters('hostingPlanName')]",
```
Comment (bmoore-msft) [inline sonarqube-webapp-linux-azuresql/azuredeploy.json]: why are we doing direct replacement?  could just use the parameter?

File: sonarqube-webapp-linux-azuresql/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+              "name": "SONARQUBE_JDBC_USERNAME",
+              "value": "[parameters('sqlUsername')]"
+            },
+            {
+              "name": "SONARQUBE_JDBC_PASSWORD",
+              "value": "[parameters('sqlPassword')]"
+            },
+            {
+              "name": "SONARQUBE_JDBC_URL",
+              "value": "[concat('jdbc:sqlserver://', parameters('sqlServerName'), '.database.windows.net;databaseName=sonar')]"
```
Comment (bmoore-msft) [inline sonarqube-webapp-linux-azuresql/azuredeploy.json]: use the reference() function to get the uri

File: sonarqube-webapp-linux-azuresql/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+              "name": "DOCKER_CUSTOM_IMAGE_NAME",
+              "value": "sonarqube"
+            }
+          ]
+        },
+        "name": "[parameters('siteName')]",
+        "serverFarmId": "[variables('hostingPlanName')]",
+        "hostingEnvironment": ""
+      },
+      "apiVersion": "2016-08-01",
```
Comment (bmoore-msft) [inline sonarqube-webapp-linux-azuresql/azuredeploy.json]: move apiVersion and location up above dependsOn

File: sonarqube-webapp-linux-azuresql/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+      "kind": "linux",
+      "tags": {
+        "displayName": "App Service Plan"
+      }
+    },
+    {
+      "name": "[parameters('sqlServerName')]",
+      "type": "Microsoft.Sql/servers",
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2014-04-01-preview",
```
Comment (bmoore-msft) [inline sonarqube-webapp-linux-azuresql/azuredeploy.json]: use the non-preview version

File: sonarqube-webapp-linux-azuresql/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+      "properties": {
+        "administratorLogin": "[parameters('sqlUsername')]",
+        "administratorLoginPassword": "[parameters('sqlPassword')]"
+      },
+      "resources": [
+        {
+          "name": "AllowAllWindowsAzureIps",
+          "type": "firewallrules",
+          "location": "[resourceGroup().location]",
+          "apiVersion": "2014-04-01-preview",
```
Comment (bmoore-msft) [inline sonarqube-webapp-linux-azuresql/azuredeploy.json]: use the non-preview version

File: sonarqube-webapp-linux-azuresql/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+          "properties": {
+            "startIpAddress": "0.0.0.0",
+            "endIpAddress": "0.0.0.0"
+          }
+        },
+        {
+          "name": "[variables('databaseName')]",
+          "type": "databases",
+          "location": "[resourceGroup().location]",
+          "apiVersion": "2014-04-01-preview",
```
Comment (bmoore-msft) [inline sonarqube-webapp-linux-azuresql/azuredeploy.json]: use the non-preview version

File: sonarqube-webapp-linux-azuresql/azuredeploy.parameters.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+      "value": "GEN-UNIQUE"
+    },
+    "hostingPlanName": {
+      "value": "sonarqube-plan"
+    },
+    "sqlServerName": {
+      "value": "GEN-UNIQUE"
+    },
+    "sqlUsername": {
+      "value": "sonarqubesqlsa"
```
Comment (bmoore-msft) [inline sonarqube-webapp-linux-azuresql/azuredeploy.parameters.json]: GEN-UNIQUE

File: sonarqube-webapp-linux-azuresql/azuredeploy.parameters.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "siteName": {
+      "value": "GEN-UNIQUE"
+    },
+    "hostingPlanName": {
+      "value": "sonarqube-plan"
```
Comment (bmoore-msft) [inline sonarqube-webapp-linux-azuresql/azuredeploy.parameters.json]: I think this needs to be unique with the subscription - so use GEN-UNIQUE here too

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**